sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace/_ai_workspaces/capability_host.py

```diff
@@ -0,0 +1,236 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import os
+from os import PathLike
+from typing import (
+    List,
+    Optional,
+    Union,
+    cast,
+    IO,
+    Any,
+    AnyStr,
+    Dict,
+)
+from pathlib import Path
+from azure.ai.ml._utils._experimental import experimental
+from azure.ai.ml.entities._resource import Resource
+from azure.ai.ml.constants._workspace import CapabilityHostKind
+from azure.ai.ml.constants._common import (
+    BASE_PATH_CONTEXT_KEY,
+    PARAMS_OVERRIDE_KEY,
+)
+
+from azure.ai.ml._schema.workspace.ai_workspaces.capability_host import (
+    CapabilityHostSchema,
+)
+from azure.ai.ml._utils.utils import dump_yaml_to_file
+from azure.ai.ml.entities._util import load_from_dict
+from azure.ai.ml._restclient.v2024_10_01_preview.models._models_py3 import (
+    CapabilityHost as RestCapabilityHost,
+)
+from azure.ai.ml._restclient.v2024_10_01_preview.models._models_py3 import (
+    CapabilityHostProperties as RestCapabilityHostProperties,
+)
+
+
+@experimental
+class CapabilityHost(Resource):
+    """Initialize a CapabilityHost instance.
+    Capabilityhost management is controlled by MLClient's capabilityhosts operations.
+
+    :param name: The name of the capability host.
+    :type name: str
+    :param description: The description of the capability host.
+    :type description: Optional[str]
+    :param vector_store_connections: A list of vector store  (AI Search) connections.
+    :type vector_store_connections: Optional[List[str]]
+    :param ai_services_connections: A list of OpenAI service connection.
+    :type ai_services_connections: Optional[List[str]]
+    :param storage_connections: A list of storage connections.
+    :type storage_connections: Optional[List[str]]
+    :param capability_host_kind: The kind of capability host, either as a string or CapabilityHostKind enum.
+    :type capability_host_kind: Optional[Union[str, CapabilityHostKind]]
+    :param kwargs: Additional keyword arguments.
+    :type kwargs: Any
```

Comment:
add sample and doc string
e.g.
 .. literalinclude:: ../samples/ml_samples_workspace.py
            :start-after: [START workspace_hub]
            :end-before: [END workspace_hub]
            :language: python
            :dedent: 8
            :caption: Creating a Hub object.
    """

Reply:
done